pkg/ebpf/bytecode/runtime/runtime_asset.go

```diff
@@ -99,19 +129,41 @@ func (a *RuntimeAsset) Compile(config *ebpf.Config, cflags []string) (CompiledOu
 	outputFile := filepath.Join(config.RuntimeCompilerOutputDir, fmt.Sprintf("%s-%d-%s-%s.o", baseName, kv, hash, flagHash))
 	if _, err := os.Stat(outputFile); err != nil {
 		if !os.IsNotExist(err) {
+			a.compilationResult = outputFileErr
 			return nil, fmt.Errorf("error stat-ing output file %s: %w", outputFile, err)
 		}
 		comp, err := compiler.NewEBPFCompiler(config.KernelHeadersDirs, config.BPFDebug)
 		if err != nil {
+			a.compilationResult = newCompilerErr
 			return nil, fmt.Errorf("failed to create compiler: %w", err)
 		}
 		defer comp.Close()
 
 		if err := comp.CompileToObjectFile(inputReader, outputFile, flags); err != nil {
+			a.compilationResult = compilationErr
 			return nil, fmt.Errorf("failed to compile runtime version of %s: %s", a.filename, err)
 		}
 	}
-	return os.Open(outputFile)
+
+	out, err := os.Open(outputFile)
+	if err == nil {
+		a.compilationResult = success
+	} else {
+		a.compilationResult = resultReadErr
+	}
+	return out, err
+}
+
+func (a *RuntimeAsset) GetTelemetry() map[string]int64 {
```

Comment:
If this function is called after compilation has started, but before compilation is completed, we will incorrectly return a `0/success` result for `runtime_compilation_result` correct? Perhaps we should add a `0` value to the enum that is `unknown`, `incomplete`, or something to indicate compilation hasn't completed yet.

Reply:
You're totally right. On top of that, I realized there's also the possibility that runtime compilation is enabled but that certain assets won't be attempted to be compiled because the tracer failed to compile, so I added a `NotAttempted` field to represent that - which should cover the use case you're describing too :)

Reply:
I haven't updated this PR yet just because I'm working on trying to fix a stubborn test case, but the agent-payload side has already been updated to reflect this